### ISSUE TITLE
Problem: user cannot upload rights metadata on transfer or sub-folder level using rights.csv import

### ISSUE THREAD

**Issue by jfcarrano** [archivematica/Issues#1340]
**Please describe the problem you'd like to be solved**
You can add PREMIS rights information via the rights.csv import at the individual file level only. You cannot add it to the full transfer or sub-folder level. Adding PREMIS rights to a full SIP is available within the Dashboard UI however. 

**Describe the solution you'd like to see implemented**
PREMIS rights.csv import should work at any level indicated by file path in the rights.csv file (full transfer, sub-folder, etc., and individual file). This will align it with the current functionality of the [metadata.csv file upload](https://www.archivematica.org/en/docs/archivematica-1.12/user-manual/transfer/import-metadata/#metadata-csv). The rights.csv upload should work similarly for BagIt transfers as well.

**Additional context**
This feature will decrease manual steps and enable users to better apply metadata to materials in aggregate as is standard in archival guidelines

---

**For Artefactual use:**

Before you close this issue, you must check off the following:

- [ ] All pull requests related to this issue are properly linked
- [ ] All pull requests related to this issue have been merged
- [ ] A testing plan for this issue has been implemented and passed (testing plan information should be included in the issue body or comments)
- [ ] Documentation regarding this issue has been written and merged
- [ ] Details about this issue have been added to the release notes

**Comment by sromkey** [comment]
Something to consider is that when you add PREMIS rights in the UI, it applies them individually to each object in the AIP (a holdover from pre-PREMIS 3, when it wasn't possible to assign rights to a directory or AIP because they are an intellectual rather than a physical entity). My assumption is that it would be preferable going forward to leverage PREMIS 3 capabilities and assign the rights to the folder or AIP level, rather than repeating throughout the METS, but @jfcarrano let us know if you think otherwise.

**Comment by jfcarrano** [comment]
That makes sense to me!

**Comment by aray-wellcome** [comment]
Hello! I was just speaking to @jhsimpson about this ticket as we would like it the rights.csv to work more in line with how the metadata.csv works as well and he asked me to put some more detail into how we're trying to use the rights.csv 

1. We're using the metadata.csv very basically and at a high level. Our metadata is applied to the entire directory like so: 
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/raya/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/raya/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">



filename | dc.identifier
-- | --
objects/ | ARTCOO/B/10



</body>

</html>

2.  Just for some background, we want to apply rights in a .csv as well but we don't want to use the PREMIS rights as you'd expect. We want to align the rights codes and access status names with our digitized work and we have found a way to do so with the following fields:
**For cc licenses**
File, basis, note, grant_act, grant_note
**For in copyright**
File, basis, status, jurisdiction, note, grant_act, grant_note

3. We want to be able to bulk apply the rights via the rights.csv in varying permutations:
**Whole directory**
Like in the metadata.csv, i.e.
File
/objects
**By subdirectory**
If we have multiple subfolders we may want to apply the rights to each subfolder
**Have a mix**
Perhaps we'd like to name the whole directory as a certain right/access status but then if we name an individual file or a subdirectory with its own right/access status, that would be applied instead of the one for the whole directory

Basically, anything to not have to make us list out every single file in the SIP and apply rights lines to every one, I think is the summary!

**Comment by alueschow** [comment]
Came across this issue when trying to upload PREMIS rights in a Bag on a folder level (e.g., `data/my_images/` as value in the `file` column of rights.csv) which does not work. The documentation is misleading, it says "Rights metadata can be applied to individual objects within a transfer, to directories within the transfer, or both." (https://www.archivematica.org/en/docs/archivematica-1.17/user-manual/transfer/import-metadata/#rights-csv)

Any progress with this issue?
